scripts/zones/Bastok_Markets/npcs/Gulldago.lua

```diff
@@ -4,16 +4,116 @@
 -- Type: Tutorial NPC
 -- !pos -364.121 -11.034 -167.456 235
 -----------------------------------
+local ID = require("scripts/zones/Bastok_Markets/IDs")
+require("scripts/globals/status")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+require("scripts/globals/npc_util")
+-----------------------------------
+
 
-function onTrade(player,npc,trade)
-end;
+function onTrade(player, npc, trade)
+end
 
-function onTrigger(player,npc)
-    player:startEvent(535);
-end;
+function onTrigger(player, npc)
+    local tutorial = player:getCharVar("TutorialProgress")
+    if tutorial == 11 then
+        player:startEvent(535)
+    else
+        local isSkilled = false
+        local mLevel = player:getMainLvl()
+        local hasCrystal = player:hasKeyItem(tpz.ki.DEM_GATE_CRYSTAL)
```

Comment:
you mean variable

variables are good when the data you are variablizing:
* changes
* gets used multiple times
* would make the line where its used wider

Since this data is static the line doesn't change and it is used only once, its cool to do it in-line.